heartbeat/monitors/factory.go

```diff
@@ -119,10 +119,6 @@ func newCommonPublishConfigs(info beat.Info, cfg *common.Config) (pipetool.Confi
 		fields := clientCfg.Processing.Fields.Clone()
 		fields.Put("event.dataset", dataset)
 
-		if settings.DataStream != nil {
```

Comment:
Think i added this as a fix for Heartbeat not recognizing the data stream when used inside agent - https://github.com/elastic/beats/pull/26774.

I hope it still works without this change?

Reply:
I just moved the logic to the `add_data_stream` processor. So, it's still running.